toolbox/dependency.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+// dependencies may take different form and shall be update accordingly
+type dependencyType uint8
+
+const (
+	/*
+		Example:
+		new_git_repository(
+			name = "mixerapi_git",
+			commit = "ee9769f5b3304d9e01cd7ed6fb1dbb9b08e96210",
+			remote = "https://github.com/istio/api.git",
+			build_file_content = BUILD,
+		)
+	*/
+	bazelRule = dependencyType(iota)
+	/*
+		Example:
+		PROXY = "466a7346f3c2c2f3cbcb0f1a74d2cd9ac6a565fe"  # Jun 2, 2017
+		http_file(
+			name = "istio_proxy",
+			url = "https://storage.googleapis.com/istio-build/proxy/envoy-alpha-" + PROXY + ".tar.gz",
+		)
+	*/
+	constant
+)
+
+type dependency struct {
+	name       string
+	repoName   string
+	prodBranch string // either master or stable
+	depType    dependencyType
+}
+
+type repoInfo struct {
+	depFile string // where in the *parent* repo such dependecy isrecorded
```

Comment:
There could be more than one file. I would move the file up to dependency as there could one file per deps.